quelpa.el

```diff
@@ -1633,13 +1632,24 @@ Return t in each case."
 
 (defun quelpa-update-cache (cache-item)
   ;; try removing existing recipes by name
-  (setq quelpa-cache (cl-remove (car cache-item)
-                                quelpa-cache :key #'car))
+  (setq quelpa-cache (--remove (eq (car cache-item) (car it))
+                               quelpa-cache))
   (push cache-item quelpa-cache)
   (setq quelpa-cache
         (cl-sort quelpa-cache #'string<
                  :key (lambda (item) (symbol-name (car item))))))
 
+(defun quelpa-remove-cache (cache-item)
+  "Remove cache-item from `quelpa-cache'."
+  (let ((cache-version (plist-get (cdr cache-item) :version)))
```

Comment:
In general, I'd recommend using `equal` instead of `eq`, even when `eq` may work in a specific case.  It will help to avoid bugs, because `equal` is nearly always correct, while `eq` often is not.